features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -167,33 +167,60 @@ class Base64Serializer private constructor(
 
     // region builder
     internal class Builder {
-        internal fun build(
-            threadPoolExecutor: ExecutorService = THREADPOOL_EXECUTOR,
-            drawableUtils: DrawableUtils = DrawableUtils(),
-            base64Utils: Base64Utils = Base64Utils(),
-            webPImageCompression: ImageCompression = WebPImageCompression(),
-            base64LruCache: Cache<Drawable, String> = Base64LRUCache,
-            bitmapPool: Cache<String, Bitmap> = BitmapPool,
-            // Temporarily use UNBOUND until we handle the loggers
-            logger: InternalLogger = InternalLogger.UNBOUND
-        ) =
+        // Temporarily use UNBOUND until we handle the loggers
+        private var logger: InternalLogger = InternalLogger.UNBOUND
+        private var threadPoolExecutor: ExecutorService = THREADPOOL_EXECUTOR
+        private var bitmapPool: BitmapPool? = null
+        private var base64LRUCache: Cache<Drawable, String>? = null
+        private var drawableUtils: DrawableUtils = DrawableUtils(bitmapPool = bitmapPool)
+        private var base64Utils: Base64Utils = Base64Utils()
+        private var webPImageCompression: ImageCompression = WebPImageCompression()
+
+        internal fun build() =
             Base64Serializer(
+                logger = logger,
                 threadPoolExecutor = threadPoolExecutor,
+                bitmapPool = bitmapPool,
+                base64LRUCache = base64LRUCache,
                 drawableUtils = drawableUtils,
                 base64Utils = base64Utils,
-                webPImageCompression = webPImageCompression,
-                base64LruCache = base64LruCache,
-                bitmapPool = bitmapPool,
-                logger = logger
+                webPImageCompression = webPImageCompression
             )
 
+        internal fun setLogger(logger: InternalLogger) {
```

Comment:
where are all these setters methods being used ?

Reply:
if they are part of the builder, than shouldn't they return the `Builder` instance ?

Reply:
these are setters provided to allow changing the builder internal state before build is called. I'm using these setters in the unit tests to set the mocks

Reply:
well in that case you need to use a `Builder` constructor with all these parameters and having default values. Meaning that for the production code you will just call the `Builder()` which takes the default ones and for testing purposes you will be able to mock any parameter.